### ISSUE TITLE
Prefix for resource names in ALM DA

### ISSUE THREAD

**Issue by in-1911** [terraform-ibm-modules/terraform-ibm-devsecops-alm#438]
### Description

The ALM DA does not have a common prefix parameter and most of the resources created by it use names explicitly provided by the inputs, but there is a parameter for repo name prefix (only applied to git repo names). As a result, if multiple stack deployments are done in the same account, the users have to go into the ALM DA inputs and manually change multiple values to use different names in the ALM resources:

- Toolchain name
- registry_namespace
- Code engine project names (CI and CD)

### New or affected modules

---

By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/terraform-ibm-modules/documentation/blob/main/CODE_OF_CONDUCT.md)

**Comment by huayuenh** [comment]
In progress

**Comment by in-1911** [comment]
It looks like rel 1.9.4 is now using the prefix in the names for toolchains and code engine projects, but the registry_namespace is still used without the prefix

**Comment by huayuenh** [comment]
There are two modes supported. 
1) Uses an existing `registry_namespace` or 2) Set a `prefix` and optionally an additionally random suffix which will create the registry namespace.

To use option 2) set 'prefix' to the value of your prefix
optionally set 'add_container_name_suffix' to 'true' 
and set 'create_icr_namespace' to true

**Comment by in-1911** [comment]
@huayuenh - could not test the `create_icr_namespace` due to #448 
When deploying without that, I think the pipeline parameter for the namespace is actually using the prefix, for some reason in some of my "upgrade" runs I did see the value without prefix.

**Comment by in-1911** [comment]
Testing with 1.9.4 again, a fresh deployment with 
```
registry_namespace = "rag-sample-app"
prefix = "rag-lab-06"
create_icr_namespace = false
```

gives:
```
# module.devsecops_ci_toolchain[0].module.pipeline_ci.ibm_cd_tekton_pipeline_property.ci_pipeline_registry_namespace will be created
 2024/07/25 17:37:16 Terraform plan |   + resource "ibm_cd_tekton_pipeline_property" "ci_pipeline_registry_namespace" {
 2024/07/25 17:37:16 Terraform plan |       + href        = (known after apply)
 2024/07/25 17:37:16 Terraform plan |       + id          = (known after apply)
 2024/07/25 17:37:16 Terraform plan |       + locked      = false
 2024/07/25 17:37:16 Terraform plan |       + name        = "registry-namespace"
 2024/07/25 17:37:16 Terraform plan |       + pipeline_id = (known after apply)
 2024/07/25 17:37:16 Terraform plan |       + type        = "text"
 2024/07/25 17:37:16 Terraform plan |       + value       = "rag-sample-app"
 2024/07/25 17:37:16 Terraform plan |     }
```

Can you confirm that this is the intended logic and the prefix is only used in the namespace when `create_icr_namespace = true`.
I would expect that if the prefix is supplied, it should be used across the board.

**Comment by huayuenh** [comment]
This was the intended behaviour as I was expecting to use it for a stack where everything is created in the same resource group. I've updated the logic to separate out the ICR logic in this PR here https://github.com/terraform-ibm-modules/terraform-ibm-devsecops-alm/pull/450. Will have a release tomorrow for it

**Comment by huayuenh** [comment]
Behaviour updated for ICR. v1.9.5 pre-release available
Optionally set the `prefix` and/or the `add_container_name_suffix`
Then when you set `create_icr_namespace` to true, this will get Terraform to create the ICR namespace, leaving it set to false, will get the CI pipeline run to create the namespace

**Comment by ocofaigh** [comment]
This is addressed in 1.9.5 right? Are we good to close this?

**Comment by in-1911** [comment]
@ocofaigh - yes we are good on this one with 1.9.5 as released to the catalog.
